addons/html_editor/static/src/main/link/link_popover.js

```diff
@@ -16,6 +16,8 @@ export class LinkPopover extends Component {
         getExternalMetaData: Function,
         isImage: Boolean,
         readOnly: { type: Boolean, optional: true },
+        fileUploadEnabled: { type: Boolean, optional: true },
+        uploadFile: { type: Function, optional: true },
```

Comment:
The name `uploadFile` kind of sounds like it could be a boolean IMO. I legit thought it was equivalent to `fileUploadEnabled` until I noticed `type: Function`. To be honest I'm not sure I understand the separation of concerns here. There is an option to enable file upload with the link popover, but it does nothing unless you also provide `uploadFile`. Couldn't the link popover handle the local upload service etc? I guess what it is missing is the record info like `resModel`, `resId`, but maybe that could be added afterwards in some kind of `uploadCallback` that would receive the new attachment? We don't want to make two writes, though.

Reply:
Indeed the popover can access the upload service and do its thing.
After the changes, these are props:
```js
        ...
        recordInfo: Object,
        canEdit: { type: Boolean, optional: true }, // from another commit
        canUpload: { type: Boolean, optional: true },
        onUpload: { type: Function, optional: true },
```
Adding the `resId` and `resModel` must be done when uploading, it cannot be done afterwards without a second write.